aw_client/client.py

```diff
@@ -84,6 +84,11 @@ def replace_last_event(self, bucket, event: Event):
         data = event.to_json_dict()
         self.dispatch_thread.add_request(endpoint, data)
 
+    def heartbeat(self, bucket, event: Event, pulsetime: float):
+        endpoint = "buckets/{}/heartbeat?pulsetime={}".format(bucket, pulsetime)
+        data = event.to_json_dict()
+        self.dispatch_thread.add_request(endpoint, data)
```

Comment:
It is not the size that matters though, it is the request count. The RequestDispatcher thread is sending requests blocking, so it will take a while. If we assume that each request takes 20ms(back and forth + aw-server processing) and it sends one event per second in 24hrs it will be 86400 requests, it will take 86400*0.0020=172.8s=~2.8min to clear the queue.

It's not an issue since the client will continue running and queueing events while the RequestDispatcher is clearing the queue, but I wouldn't call it an microoptimization. If we later make it possible to have a remote aw-server the response time could be much higher and become a bigger issue, but still not critical.